extensions/wikia/EmbeddableDiscussions/scripts/sharing.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Functionality for social network sharing for Embeddable Discussions extension
+ */
+define('EmbeddableDiscussionsSharing',
+	['jquery'],
+	function ($) {
+		'use strict';
+
+		var socialNetworks = {
+			en: [
+				'facebook',
+				'twitter',
+				'reddit',
+				'tumblr'
+			],
+			ja: [
+				'facebook',
+				'twitter',
+				'google',
+				'line'
+			],
+			'pt-br': [
+				'facebook',
+				'twitter',
+				'reddit',
+				'tumblr'
+			],
+			zh: [
+				'facebook',
+				'weibo'
+			],
+			de: [
+				'facebook',
+				'twitter',
+				'tumblr'
+			],
+			fr: [
+				'facebook',
+				'twitter'
+			],
+			es: [
+				'facebook',
+				'twitter',
+				'meneame',
+				'tumblr'
+			],
+			ru: [
+				'vkontakte',
+				'facebook',
+				'odnoklassniki',
+				'twitter'
+			],
+			pl: [
+				'facebook',
+				'twitter',
+				'nk',
+				'wykop'
+			]
+		};
+
+		function getShareUrl(network, sharedUrl, sharedTitle) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 26 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)